src/lib/services/access-service.ts

```diff
@@ -351,8 +351,12 @@ export class AccessService {
         return this.store.getRolesForUserId(userId);
     }
 
-    async unlinkUserRoles(userId: number): Promise<void> {
-        return this.store.unlinkUserRoles(userId);
+    async wipeUserPermissions(userId: number): Promise<void> {
+        await this.store.unlinkUserRoles(userId);
+        await this.store.unlinkUserGroups(userId);
+        await this.store.clearUserPersonalAccessTokens(userId);
+        await this.store.clearPublicSignupUserTokens(userId);
+        return Promise.resolve();
```

Comment:
Why do we need this? An async function returns a promise by default.

Also: We are making 4 calls here. Could we do them in parallel? Something like:

```
const actions = [this.store.unlinkUserRoles, this.store.unlinkUserGroups, this.store.clearPersonaAccessTokens, this.store.clearPublichSignupUser]

await Promise.all(actions.map(action => action(userId)))
```

Reply:
Made it parallel, but I am not sure about this mapping. It definitely does not help with readability and does not make it much shorter.